src/ucp/core/ucp_request.h

```diff
@@ -183,15 +183,14 @@ struct ucp_request {
                         ucp_tag_t tag;
 
                         struct {
-                            union {
-                                /* Can be union, because once header is packed to
-                                 * reg_desc, it is not accessed anymore. */
-                                void           *header;
+                            struct {
+                                void           *usr_ptr;
```

Comment:
Maybe it's better to use "user_ptr" instead of the "usr_ptr" so it will be more clear.
In my latest PR I was advised to do so when I used "usr" instead of "user"